src/EventStore.Core/Helpers/IODispatcherDelayedMessage.cs

```diff
@@ -10,22 +10,22 @@ public sealed class IODispatcherDelayedMessage : Message {
 		}
 
 		private readonly Guid _correlationId;
-		private readonly Action _action;
+		private readonly ICorrelatedTimeout _timeout;
 		private readonly Guid? _messageCorrelationId;
 
-		public IODispatcherDelayedMessage(Guid correlationId, Action action) {
-			_action = action;
+		public IODispatcherDelayedMessage(Guid correlationId, ICorrelatedTimeout timeout) {
```

Comment:
suggestion: rename `IODispatcherDelayedMessage` to `IODispatcherTimeoutMessage`

previously the code was handling a general, scheduled message, so the name `IODispatcherDelayedMessage` was appropriate but now it handles timeouts only